pkg/experiment/ingester/bucket_mock_test.go

```diff
@@ -0,0 +1,72 @@
+package ingester
+
+import (
+	"context"
+	objstore2 "github.com/grafana/pyroscope/pkg/objstore"
+	"github.com/thanos-io/objstore"
+	"io"
+)
+
+type mockBucket struct {
+	upload func(ctx context.Context, name string, r io.Reader) error
+}
```

Comment:
Super nit: `github.com/stretchr/testify/mock` is very convenient – if you find yourself writing yet another mock, give it a look. It can also generate boilerplate for you https://vektra.github.io/mockery/latest/

Reply:
Let's merge it as is now and I will create a followup PR with testify/mock